### PR TITLE
Create data providers

### PR DESCRIPTION
**Pull request by emanrdesu**

### CODE REVIEW

File: java.exercises/src/test/java/DataProviderTests.java [hunk 1]
```diff
@@ -0,0 +1,72 @@
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+public class DataProviderTests {
+	@Test
+	public void canGetStateName() {
+		var expectedState = "Texas";
```
Comment (dentaylor) [inline java.exercises/src/test/java/DataProviderTests.java:7]: Insert a line between the three sections (Arrange, Act, Assert)

File: java.exercises/src/test/java/LocationBuilder.java [hunk 2]
```diff
@@ -0,0 +1,23 @@
+
+public class LocationBuilder {
+
+	private LocationObjectMother mother;
+
+	public LocationBuilder() {
+		this.mother = new LocationObjectMother();
+	}
+
+	public LocationBuilder withState(State s) {
```
Comment (dentaylor) [inline java.exercises/src/test/java/LocationBuilder.java:10]: Don't use abbreviations as parameter names. Spell it out 
public LocationBuilder withState(State state);

**Review by dentaylor — CHANGES_REQUESTED** [review]

File: java.exercises/src/test/java/DataProviderTests.java [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+		
+		var actualState = States.Texas.Name;
+		
+		var errorMessage = "State name should be returned.";
+		Assert.assertEquals(actualState, expectedState, errorMessage);
```
Comment (dentaylor) [inline java.exercises/src/test/java/DataProviderTests.java:12]: Suggest inlining the error message since you're only using it once.

**Review by dentaylor — APPROVED** [review]

File: java.exercises/src/test/java/DataProviderTests.java (hunk 3 above)
Comment (emanrdesu) [reply java.exercises/src/test/java/DataProviderTests.java:12]: I know, but I don't like lines being very long, is it ok if I leave as is?

**Pull request merged**